packages/multichain/src/caip25Permission.test.ts

```diff
@@ -153,8 +153,51 @@ describe('caip25EndowmentBuilder', () => {
         });
       });
 
+      it('revokes the permission if the only non wallet scope is removed', () => {
+        const caveatValue = {
+          requiredScopes: {},
+          optionalScopes: {
+            'eip155:5': {
+              accounts: [],
+            },
+            'wallet:eip155': {
+              accounts: [],
+            },
+            wallet: {
+              accounts: [],
+            },
+          },
+          sessionProperties: {},
+          isMultichainOrigin: true,
+        };
+        const result = removeScope(caveatValue, 'eip155:5');
+        expect(result).toStrictEqual({
+          operation: CaveatMutatorOperation.RevokePermission,
+        });
+      });
+
+      it('does not revoke the permission if the target scope does not exist but the permission only has wallet scopes', () => {
```

Comment:
Nit: The "does not revoke" tests are a bit confusing because it's not clear what we do expect to happen. Could we re-phrase them to make clear that we expect the result to be a no-op? The same goes for those added later in this file as well

Reply:
Actually maybe these cases are already covered? I do see some "account does not exist" and "scope does not exist" test cases already.

Reply:
done here https://github.com/MetaMask/core/pull/4978/commits/d65e8eb284364cc4773fdef305e013228f6ce239